patches/tModLoader/Terraria/DataStructures/PlayerDeathReason.cs.patch

```diff
@@ -63,10 +66,40 @@
  			return true;
  		}
  
-@@ -72,18 +_,26 @@
+@@ -65,6 +_,10 @@
  		};
  	}
  
++	/// <summary>
++	/// A custom death message.
++	/// <para/> This overload does not support localization, all players will see the same text regardless of their language settings. Use <see cref="ByCustomReason(LocalizedText)"/> or <see cref="ByCustomReason(NetworkText)"/> instead for localization support.
++	/// </summary>
+ 	public static PlayerDeathReason ByCustomReason(string reasonInEnglish)
+ 	{
+ 		return new PlayerDeathReason {
+@@ -72,18 +_,45 @@
+ 		};
+ 	}
+ 
++	/// <inheritdoc cref="ByCustomReason(NetworkText)"/>
++	public static PlayerDeathReason ByCustomReason(LocalizedText reason)
++	{
++		return new PlayerDeathReason {
++			_sourceCustomReasonLocalized = reason.ToNetworkText()
++		};
++	}
++
++	/// <summary>
++	/// A localized custom death message. Each client will see the death message in their own language, unlike with <see cref="ByCustomReason(string)"/>.
++	/// <para/> Text substitutions can be provided by using <see cref="LocalizedText.ToNetworkText(object[])"/> or <see cref="LocalizedText.WithFormatArgs(object[])"/>
```

Comment:
Don't mention `WithFormatArgs` here. Standard ways of creating `NetworkText` are all good